app/javascript/packs/building_map.js

```diff
@@ -1,70 +1,86 @@
-import { displayRoute, setupMap } from './leafletMap.js';
+import { displayRoute, setupMap, pins } from './leafletMap.js';
 
 let currentLocation;
 const YOUR_LOCATION_MAGIC_STRING = "Your location" // This will be changed when the page supports multiple languages
+const PIN_1_MAGIC_STRING = "Pin 1"
+const PIN_2_MAGIC_STRING = "Pin 2"
 
 setupMap();
 
-const start_input_field = $("#start_input")[0];
-start_input_field.addEventListener("change", () => {
-    request_location();
-    if (validate_place_input("start_input", "startOptions")) {
-        start_input_field.setCustomValidity("");
+const startInputField = $("#startInput")[0];
+startInputField.addEventListener("change", () => {
+    requestLocation();
+    if (validatePlaceInput("startInput", "startOptions")) {
+        startInputField.setCustomValidity("");
     } else {
-        start_input_field.setCustomValidity(
+        startInputField.setCustomValidity(
             "Please select a valid starting place."
         );
     }
 });
-start_input_field.dispatchEvent(new Event("change"));
+startInputField.dispatchEvent(new Event("change"));
 
-const dest_input_field = $("#dest_input")[0];
-dest_input_field.addEventListener("change", () => {
-    if (validate_place_input("dest_input", "destOptions")) {
-        dest_input_field.setCustomValidity("");
+const destInputField = $("#destInput")[0];
+destInputField.addEventListener("change", () => {
+    if (validatePlaceInput("destInput", "destOptions")) {
+        destInputField.setCustomValidity("");
     } else {
-        dest_input_field.setCustomValidity(
+        destInputField.setCustomValidity(
             "Please select a valid destination place."
         );
     }
 });
-dest_input_field.dispatchEvent(new Event("change"));
+destInputField.dispatchEvent(new Event("change"));
 
-$("#navigation_form")[0]
+$("#navigationForm")[0]
     .addEventListener("submit", (event) => {
         event.preventDefault();
-        if (start_input_field.value === YOUR_LOCATION_MAGIC_STRING) {
-            start_input_field.value = currentLocation;
+        for(let inputField of [startInputField, destInputField]){
```

Comment:
In my opinion it might be more sensible to not replace these with the coordinates but just use the coordinates for route calculation.

Reply:
The replacement was necessary when we reloaded the page for route calculation (because we somehow had to send the current location/pin location), but as we basically have a SPA now, it might really be better to do it the way Julian suggested.

Reply:
Good point, we changed it